include/unifex/let_with.hpp

```diff
@@ -0,0 +1,183 @@
+        (requires is_error_receiver_v<Receiver, Error>)
+    void set_error(Error&& error) noexcept {
+        unifex::set_error(std::move(receiver_), (Error&&)error);
+    }
+
+    template(typename R = Receiver)
+        (requires is_done_receiver_v<R>)
+    void set_done() noexcept {
+        unifex::set_done(std::move(receiver_));
+    }
+
+    template(typename CPO, typename Self)
+        (requires
+            is_receiver_query_cpo_v<CPO> AND
+            same_as<Self, type>)
+    friend auto tag_invoke(CPO cpo, const Self& self)
+        noexcept(is_nothrow_callable_v<CPO, const Receiver&>)
+        -> callable_result_t<CPO, const Receiver&> {
+        return cpo(self.receiver_);
+    }
+
+private:
+    UNIFEX_NO_UNIQUE_ADDRESS Operation& op_;
+    UNIFEX_NO_UNIQUE_ADDRESS Receiver receiver_;
+};
+
+template<typename StateFactory, typename SuccessorFactory>
+struct _let_with_sender {
+    class type;
+};
+
+template<typename StateFactory, typename SuccessorFactory>
+using let_with_sender = typename _let_with_sender<StateFactory, SuccessorFactory>::type;
+
+template<typename StateFactory, typename SuccessorFactory>
+class _let_with_sender<StateFactory, SuccessorFactory>::type {
+public:
+    using InnerOp = std::invoke_result_t<SuccessorFactory, std::invoke_result_t<StateFactory>&>;
+
+    template<template<typename...> class Variant, template<typename...> class Tuple>
+    using value_types = typename InnerOp::template value_types<Variant, Tuple>;
+
+    template<template<typename...> class Variant>
+    using error_types = typename InnerOp::template error_types<Variant>;
+
+    template<typename StateFactory2, typename SuccessorFactory2>
+    explicit type(StateFactory2&& state_factory, SuccessorFactory2&& func) :
+        state_factory_((StateFactory2&&)state_factory),
+        func_((SuccessorFactory2&&)func)
+    {}
+
+    template(typename Self, typename Receiver)
+        (requires same_as<remove_cvref_t<Self>, type>)
+    friend auto tag_invoke(tag_t<unifex::connect>, Self&& self, Receiver&& r)
+        noexcept(
+            std::is_nothrow_invocable_v<
+                member_t<Self, SuccessorFactory>,
+                std::invoke_result_t<StateFactory>&> &&
+            std::is_nothrow_constructible_v<remove_cvref_t<Receiver>, Receiver>) {
+
+        return operation<
+                member_t<Self, StateFactory>, member_t<Self, SuccessorFactory>, Receiver>(
+            static_cast<Self&&>(self).state_factory_,
+            static_cast<Self&&>(self).func_,
+            static_cast<Receiver&&>(r));
+    }
+
+private:
+    UNIFEX_NO_UNIQUE_ADDRESS StateFactory state_factory_;
+    UNIFEX_NO_UNIQUE_ADDRESS SuccessorFactory func_;
+};
+
+
+template<typename StateFactory, typename SuccessorFactory, typename Receiver>
+struct _let_with_operation<StateFactory, SuccessorFactory, Receiver>::type {
+    type(StateFactory&& state_factory, SuccessorFactory&& func, Receiver&& r) :
+        state_{state_factory()},
```

Comment:
Should be:
```
state_(static_cast<StateFactory&&>(state_factory)())
```
to be consistent with value-category of invocation of `func` below.